kiva/agg/src/agg_typemaps.i

```diff
@@ -337,3 +337,34 @@
     
     $1 = stops;
 }
+
+%typemap(in) const char* gradient_arg (PyObject *utfstr=NULL) {
+  if (PyBytes_Check($input))
+  {
+    $1 = (char *)PyBytes_AsString($input);
+  }
+#if PY_VERSION_HEX >= 0x03030000
+  else if (PyUnicode_Check($input))
+  {
+    $1 = (char *)PyUnicode_AsUTF8($input);
+  }
+#elif PY_VERSION_HEX < 0x03000000
+  else if (PyUnicode_Check($input))
+  {
+    utfstr = PyUnicode_AsUTF8String($input);
+    $1 = (char *)PyString_AsString(utfstr);
+  }
+#endif
+  else
+  {
+    PyErr_SetString(PyExc_TypeError, "not a string");
+    return NULL;
+  }
+}
+%typemap(freearg) const char* gradient_arg
+{
+    if (utfstr$argnum)
+    {
+        Py_DECREF(utfstr$argnum);
+    }
+}
```

Comment:
@transfix Can you give your opinion here whether I'm crossing my t's and dotting my i's w.r.t. Python memory management?

Reply:
[PyUnicode_AsUTF8String](https://docs.python.org/2.7/c-api/unicode.html#c.PyUnicode_AsUTF8String) returns a new string reference. If my reading of the C-API documentation is correct, this means that it's this code's responsibility to decrement the reference .

Reply:
Checks out to me. One suggestion, if possible, is to change the c string args of linear_gradient to const std::string& and use std_string.i that comes with swig.

Reply:
Thanks for looking!